x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -24,6 +24,21 @@ export interface EndpointAction {
   };
 }
 
+export interface EndpointActionResponse {
```

Comment:
Does this already exist anywhere else? @ashokaditya maybe in your PR?

Reply:
Might exist in Ash's draft. Does not exist anywhere else yet, since we don't really know what the success structure looks like (or even failure when it comes from the endpoint).

Reply:
yeah, I was wondering that as well. Feels like the Agent should know what an error/success looks like since they are the ones writing this to es. (just by looking at the schema, I thought that if `error` was populated, then it "failed", else, it was "success" 🤷‍♂️ )

Reply:
Agent knows about the fields _it_ writes. It does not know about the fields Endpoint writes (and vice versa). 

AFAIK it treats the response given via endpoint as a total unparsed black box, wraps it with its own fields, and then passes to fleet server (where maybe more fields are written? no? who knows?).

that `error` field is coming from either fleet-server or agent who have a failure in their system before being able to deliver to endpoint.